homeassistant/components/bangolufsen/media_player.py

```diff
@@ -0,0 +1,777 @@
+    ATTR_MEDIA_EXTRA,
+    BrowseMedia,
+    MediaPlayerDeviceClass,
+    MediaPlayerEntity,
+    MediaPlayerEntityFeature,
+    MediaPlayerState,
+    MediaType,
+    RepeatMode,
+    async_process_play_media_url,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_MODEL
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util.dt import utcnow
+
+from .const import (
+    ART_SIZE_ENUM,
+    BANGOLUFSEN_MEDIA_TYPE,
+    BANGOLUFSEN_STATES,
+    CONF_BEOLINK_JID,
+    CONF_DEFAULT_VOLUME,
+    CONF_MAX_VOLUME,
+    CONF_VOLUME_STEP,
+    DOMAIN,
+    ENTITY_ENUM,
+    FALLBACK_SOURCES,
+    HIDDEN_SOURCE_IDS,
+    REPEAT_ENUM,
+    SOURCE_ENUM,
+    VALID_MEDIA_TYPES,
+    WEBSOCKET_NOTIFICATION,
+)
+from .entity import BangOlufsenEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+BANGOLUFSEN_FEATURES = (
+    MediaPlayerEntityFeature.PAUSE
+    | MediaPlayerEntityFeature.SEEK
+    | MediaPlayerEntityFeature.VOLUME_SET
+    | MediaPlayerEntityFeature.VOLUME_MUTE
+    | MediaPlayerEntityFeature.PREVIOUS_TRACK
+    | MediaPlayerEntityFeature.NEXT_TRACK
+    | MediaPlayerEntityFeature.PLAY_MEDIA
+    | MediaPlayerEntityFeature.VOLUME_STEP
+    | MediaPlayerEntityFeature.SELECT_SOURCE
+    | MediaPlayerEntityFeature.STOP
+    | MediaPlayerEntityFeature.CLEAR_PLAYLIST
+    | MediaPlayerEntityFeature.PLAY
+    | MediaPlayerEntityFeature.SHUFFLE_SET
+    | MediaPlayerEntityFeature.BROWSE_MEDIA
+    | MediaPlayerEntityFeature.REPEAT_SET
+    | MediaPlayerEntityFeature.TURN_OFF
+)
+
+
+PARALLEL_UPDATES = 0
+SCAN_INTERVAL = timedelta(minutes=2)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up a Media Player entity from config entry."""
+    entity = hass.data[DOMAIN][config_entry.unique_id][ENTITY_ENUM.MEDIA_PLAYER]
+    # Add MediaPlayer entity
+    async_add_entities(new_entities=[entity], update_before_add=True)
+
+
+class BangOlufsenMediaPlayer(MediaPlayerEntity, BangOlufsenEntity):
+    """Representation of a media player."""
+
+    _attr_has_entity_name = False
+    _attr_icon = "mdi:speaker-wireless"
+    _attr_supported_features = BANGOLUFSEN_FEATURES
+
+    def __init__(self, entry: ConfigEntry) -> None:
+        """Initialize the media player."""
+        super().__init__(entry)
+
+        self._beolink_jid: str = self.entry.data[CONF_BEOLINK_JID]
+        self._default_volume: int = self.entry.data[CONF_DEFAULT_VOLUME]
+        self._max_volume: int = self.entry.data[CONF_MAX_VOLUME]
+        self._model: str = self.entry.data[CONF_MODEL]
+        self._volume_step: int = self.entry.data[CONF_VOLUME_STEP]
+
+        self._attr_device_class = MediaPlayerDeviceClass.SPEAKER
```

Comment:
Move this to class level

Reply:
I can't move the self._model because it depends on the entry data and self._attr_device_class would cause the `BangOlufsenEntity` device class to be applied instead.